nexus/src/app/rack.rs

```diff
@@ -724,4 +716,51 @@ impl super::Nexus {
 
         Ok(result)
     }
+
+    /// Return the list of sleds that are inserted into an initialized rack
+    /// but not yet initialized as part of a rack.
+    pub(crate) async fn uninitialized_sled_list(
+        &self,
+        opctx: &OpContext,
+    ) -> ListResultVec<UninitializedSled> {
+        // Grab the SPs from the last collection
+        let limit = NonZeroU32::new(50).unwrap();
```

Comment:
Can we fail more gracefully if this limit is wrong for some reason?  Like should we produce a 500 if we actually get `limit` results back?

Reply:
I believe we do fail gracefully if the limit is reached here. We pass this limit into `inventory_collection_read_all_or_nothing` which ends up [calling `bail_unless` if the limit is reached](https://github.com/oxidecomputer/omicron/pull/4504/files#diff-5d05ec64d061b45e7c1e131a7914eb2213e423e5ca75a0c6c5983a6e7721f462R838)